app/utils/documents.server.ts

```diff
@@ -496,7 +496,7 @@ async function fetchApiContentsRemote(
     },
   }
   const res = await fetch(
-    `https://api.github.com/repos/${repo}/contents/${startingPath}?=${branch}`,
+    `https://api.github.com/repos/${repo}/contents/${startingPath}?ref=${branch}`,
```

Comment:
This is what caused `index.tsx` to show up even though the actual file in the repo was `index.js`.